src/parser/ToCollector.js

```diff
@@ -46,21 +64,29 @@ ToCollector.enterParticipant = onParticipant;
 let onTo = function (ctx) {
   if (isBlind) return;
   let participant = ctx.getFormattedText();
-  participants.Add(participant);
+  const participantInstance = participants.Get(participant);
+
+  // Skip adding participant position if label is present
+  if (participantInstance?.label) {
+    participants.Add(participant, false);
+  } else {
+    participants.Add(participant, false, ctx.start.start, ctx.stop.stop + 1);
+  }
 };
 
 ToCollector.enterFrom = onTo;
 ToCollector.enterTo = onTo;
 
 ToCollector.enterStarter = function (ctx) {
   let participant = ctx.getFormattedText();
-  participants.Add(participant, true);
+  participants.Add(participant, true, ctx.start.start, ctx.stop.stop + 1);
 };
 
 ToCollector.enterCreation = function (ctx) {
   if (isBlind) return;
   const participant = ctx.Owner();
-  participants.Add(participant);
+  const ctor = ctx.creationBody().construct();
```

Comment:
`ctor` could be null when the user is typing.
For example, the user pauses at
```
A.method {
  new 
}
```
In such case, the diagram is rendered as
<img width="347" alt="image" src="https://github.com/mermaid-js/zenuml-core/assets/169417/fbb44bfe-543d-467e-8f35-2676b20dd3d9">

Reply:
Fixed by adding optional chaining.